### ISSUE TITLE
Implicit casting using jQuery's .data causes problem with sorting columns

### ISSUE THREAD

**Issue by martinberoiz** [nadangergeo/RWD-Table-Patterns#55]
Hello,

The jQuery [.data](https://github.com/nadangergeo/RWD-Table-Patterns/blob/9241a5f3995ee07f46b5e8a640e395af4bc528a7/src/js/rwd-table.js#L874) method will try to cast `data-value` into a number (or any other type) whenever possible.

From [the docs:](https://api.jquery.com/data/)

> Every attempt is made to convert the attribute's string value to a JavaScript value (this includes booleans, numbers, objects, arrays, and null). A string is only converted to a number if doing so doesn't change its representation (for example, the string "100" is converted to the number 100, but "1E02" and "100.000" are left as strings because their numeric value of 100 serializes to "100"). When a string starts with '{' or '[', then jQuery.parseJSON is used to parse it; it must follow [valid JSON syntax](https://en.wikipedia.org/wiki/JSON#Data_types.2C_syntax_and_example) including quoted property names. A string not parseable as a JavaScript value is not converted.
>
> To retrieve a data-* attribute value as an unconverted string, use the [attr()](https://api.jquery.com/attr/) method.

This has the problem that when using `data-value` to set the value of a cell, it will crash when trying to sort, because the [default comparison](https://github.com/nadangergeo/RWD-Table-Patterns/blob/9241a5f3995ee07f46b5e8a640e395af4bc528a7/src/js/rwd-table.js#L141) is made using `String.prototype.localeCompare`. Numbers don't have `localeCompare`, so it breaks.

I think it would be best to use `.attr` that has no unintended castings and always returns a string, consistent with the default comparison function.

We noticed this issue in our production code, we already prepared a patch, but I think it could help other people if we create a pull request upstream.

Would you be interested in me opening a PR with this small change?

**Comment by martinberoiz** [comment]
Here is a minimal working example (sort of).

```html
<!DOCTYPE html>
<html>
  <head>
    <!-- jQuery -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
    <!-- Bootstrap 5 -->
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-4bw+/aepP/YC94hEpVNVgiZdgIC5+VKNBQNGCHeKRQN+PtmoHDEXuppvnDJzQIu9" crossorigin="anonymous">
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.1/dist/js/bootstrap.bundle.min.js" integrity="sha384-HwwvtgBNo3bZJJLYd8oVXjrBZt8cqVSpeBNS5n7C8IVInixGAoxmnlMuBnhbgrkm" crossorigin="anonymous"></script>
     <script src="js/rwd-table.js"></script>
    <link rel="stylesheet" href="css/rwd-table.css" />
  </head>
  <body>
      <h1>Example of problem columns</h1>
      <div data-responsive-table-toolbar="my-table"></div>
      <div
        class="table-responsive"
        data-pattern="priority-columns"
        data-sortable="true"
      >
        <table class="table table-bordered table-striped" id="my-table">
          <thead>
            <tr>
              <th>Value</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td data-value="3">$3M</td>
            </tr>
            <tr>
              <td data-value="1">$1M</td>
            </tr>
            <tr>
              <td data-value="2">$2M</td>
            </tr>
            <tr>
              <td data-value="4">$4M</td>
            </tr>
            <tr>
              <td data-value="5">$5M</td>
            </tr>
          </tbody>
        </table>
      </div>
  </body>
</html>
```

**Comment by nadangergeo** [comment]
Nice catch and fix! :raised_hands:
Fixed by #56
